common/src/Utilities/PathUtils.cpp

```diff
@@ -125,72 +112,83 @@ wxString Path::Normalize(const wxDirName &src)
     return wxDirName(src).Normalize().ToString();
 }
 
-wxString Path::MakeAbsolute(const wxString &src)
+std::string Path::MakeAbsolute(const std::string &src)
 {
-    wxFileName absolute(src);
-    absolute.MakeAbsolute();
-    return absolute.GetFullPath();
+    return ghc::filesystem::absolute(src);
 }
 
 // Concatenates two pathnames together, inserting delimiters (backslash on win32)
 // as needed! Assumes the 'dest' is allocated to at least g_MaxPath length.
 //
-wxString Path::Combine(const wxString &srcPath, const wxString &srcFile)
+fs::path Path::Combine(fs::path &srcPath, fs::path &srcFile)
 {
-    return (wxDirName(srcPath) + srcFile).GetFullPath();
+    return (srcPath / srcFile).make_preferred();
 }
 
-wxString Path::Combine(const wxDirName &srcPath, const wxFileName &srcFile)
+std::string Path::Combine(const std::string &srcPath, const std::string &srcFile)
 {
-    return (srcPath + srcFile).GetFullPath();
+    fs::path srcP = srcPath;
+    fs::path srcF = srcFile;
+    return (srcP / srcF).make_preferred();
 }
-
-wxString Path::Combine(const wxString &srcPath, const wxDirName &srcFile)
-{
-    return (wxDirName(srcPath) + srcFile).ToString();
-}
-
 // Replaces the extension of the file with the one given.
 // This function works for path names as well as file names.
-wxString Path::ReplaceExtension(const wxString &src, const wxString &ext)
+std::string Path::ReplaceExtension(const wxString &src, const wxString &ext)
 {
     wxFileName jojo(src);
     jojo.SetExt(ext);
-    return jojo.GetFullPath();
+    return std::string(jojo.GetFullPath().ToUTF8());
 }
 
-wxString Path::ReplaceFilename(const wxString &src, const wxString &newfilename)
+std::string Path::ReplaceFilename(const wxString &src, const wxString &newfilename)
 {
     wxFileName jojo(src);
     jojo.SetFullName(newfilename);
-    return jojo.GetFullPath();
+    return std::string(jojo.GetFullPath().ToUTF8());
 }
 
-wxString Path::GetFilename(const wxString &src)
+std::string Path::GetFilename(const std::string &src)
 {
-    return wxFileName(src).GetFullName();
+    return ghc::filesystem::absolute(src);
 }
```

Comment:
What's up with this function?  It's not returning the filename anymore, instead its returning the absolute path?